sql/core/src/main/scala/org/apache/spark/sql/DataFrameNaFunctions.scala

```diff
@@ -419,7 +428,7 @@ final class DataFrameNaFunctions private[sql](df: DataFrame) {
           case v: jl.Boolean => fillCol[Boolean](f, v.booleanValue())
           case v: String => fillCol[String](f, v)
         }
-      }.getOrElse(df.col(f.name))
+      }.getOrElse(df.col(s"`${f.name}`"))
```

Comment:
does `replace0` have the same issue?

Reply:
Yes, should be fixed in the same PR or separate JIRA-PR?

Reply:
let's fix it with a separated JIRA